test/command/test-check-performance-regression.rb

```diff
@@ -0,0 +1,272 @@
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+require "groonga-query-log/command/check-performance-regression"
+
+class CheckPerformanceRegressionCommandTest < Test::Unit::TestCase
+  include Helper::Path
+
+  def setup
+    @command = GroongaQueryLog::Command::CheckPerformanceRegression.new
+  end
+
+  def fixture_path(*components)
+    super("check-performance-regression", *components)
+  end
+
+  sub_test_case("options") do
+
+    MISSING_QUERY_LOG_ERROR = <<-OUTPUT
+query logs is not specified. use --input-old-query and --input-new-query
+    OUTPUT
+
+    def run_command_with_stderr
+      actual = ""
+      Tempfile.open("redirect-stderr-tmpfile") do |file|
+        $stderr.reopen(file)
+        yield(file)
+        $stderr.flush
+        file.rewind
+        actual = file.read
+      end
+      actual
+    end
+
+    def test_no_option
+      actual = run_command_with_stderr do
+        @command.run([])
+      end
+      assert_equal(MISSING_QUERY_LOG_ERROR, actual)
+    end
+
+    def test_no_input_old_query
+      actual = run_command_with_stderr do
+        @command.run([
+          "--input-new-query=" + fixture_path("query2.log")
+        ])
+      end
+      assert_equal(MISSING_QUERY_LOG_ERROR, actual)
+    end
+
+    def test_no_input_new_query
+      actual = run_command_with_stderr do
+        @command.run([
+          "--input-old-query=" + fixture_path("query1.log")
+        ])
+      end
+      assert_equal(MISSING_QUERY_LOG_ERROR, actual)
+    end
+
+    def test_nonexistent_input_old_query
+      expected = <<-OUTPUT
+invalid option: --input-old-query=#{fixture_path("nonexsistent.log")}
+      OUTPUT
+      actual = run_command_with_stderr do
+        @command.run([
+          "--input-old-query=" + fixture_path("nonexsistent.log")
+        ])
+      end
+      assert_equal(expected, actual)
+    end
+
+    def test_nonexistent_input_new_query
+      expected = <<-OUTPUT
+invalid option: --input-new-query=#{fixture_path("nonexsistent.log")}
+      OUTPUT
+      actual = run_command_with_stderr do
+        @command.run([
+          "--input-new-query=" + fixture_path("nonexsistent.log")
+        ])
+      end
+      assert_equal(expected, actual)
+    end
+
+    def test_output
+      path = "/tmp/output"
+      @command.run([
+        "--input-old-query=" + fixture_path("query1.log"),
+        "--input-new-query=" + fixture_path("query2.log"),
+        "--slow-response-ratio=0",
+        "--slow-operation-ratio=0",
+        "--slow-response-threshold=0",
+        "--slow-operation-threshold=0",
+        "--output=#{path}"
+      ])
+      expected = <<-OUTPUT
+Query: select --table Site --limit 0
+  Before(average): 12000000 (nsec) After(average): 14000000 (nsec) Ratio: (+16.67% +0.00sec/+2.00msec/+2000.00usec/+2000000.00nsec)
```

Comment:
Aligned output will be easy to understand:

```text
   Before(average): 12000000 (nsec)
    After(average): 14000000 (nsec)
             Ratio: (+16.67% +0.00sec/+2.00msec/+2000.00usec/+2000000.00nsec)
```

Reply:
It seems reasonable, I've fixed.